### PR TITLE
[TAO-2511][Feature] Support frozen DINOv3 backbones in DEFT AOI

### PR DESCRIPTION
**Pull request by nnagrajrao**
### What changes are proposed in this pull request?

Extend the existing DEFT AOI workflow and generated Visual ChangeNet schemas for frozen DINOv3 while preserving the existing C-RADIO default.

- Adds staging profiles for ViT-S, S+, B, L, H+, and 7B.
- Infers the selected profile from the existing workspace baseline YAML when no CLI override is supplied.
- Hard-stops if a DINOv3 DEFT baseline is not frozen, including explicit `--backbone-type` selection.
- Reuses an explicitly configured converted TAO checkpoint before considering Hugging Face; a missing explicit checkpoint hard-stops.
- Uses the matching staged Hugging Face artifact only when no checkpoint is configured.
- Preserves the established C-RADIO host/container filename mapping, including workspaces containing other staged weights.
- Carries exact backbone type, frozen state, host artifact, and container path into the inference handoff.
- Mounts the selected host backbone at that exact container path for training and handoff inference.
- Adds ViT-7B only to Visual ChangeNet schemas; SegFormer remains unchanged.

### Why are the changes needed?

The DEFT AOI loop must run task-level fine-tuning with a frozen DINOv3 backbone, using either customer-converted TAO SSL weights or compatible public timm/Hugging Face weights, without changing the existing AOI workflow for C-RADIO users.

### Related issues

JIRA: TAO-2511.

Companion tao-pytorch and tao-core branches: `TAO-2511-frozen-dinov3-aoi`.

### Does this PR introduce _any_ user-facing change?

Yes. `stage_backbone.py` accepts all six Visual ChangeNet DINOv3 types and the existing workspace-spec-driven preflight selects the configured frozen DINO profile. C-RADIO remains the default and is backward compatible.

### How was this patch tested?

- Required TAO image, DEFT AOI application and schema-drift suites: **33 passed**.
- Full `./scripts/validate-skills.sh`: passed.
- Exact PR-range hooks: license header passed.
- Python compilation and `git diff --check`: passed.
- All six public timm/Hugging Face profile repositories were verified to contain `model.safetensors`.
- Independent architecture, correctness, and MR-readiness panel completed with no remaining blockers.

### Was this patch authored or co-authored using generative AI tooling?

Generated-by: OpenAI Codex

### Release note

```release-note
DEFT AOI now supports frozen DINOv3 ViT-S, S+, B, L, H+, and 7B backbones using explicit converted TAO checkpoints or staged timm/Hugging Face weights while preserving the C-RADIO default.
```

### Checklist

- [x] My commits are signed off per the DCO (`git commit -s`)
- [x] I have read the contributing guidelines
- [x] The code follows the project style, and repository checks pass locally
- [x] I added or updated tests covering this change
- [x] All applicable tests pass locally
- [x] Generated schemas are updated
- [x] Necessary executable skill references are updated
- [ ] A version or changelog update is not required for this patch
- [x] No secrets, credentials, proprietary data, or customer data are included
- [x] I understand the contribution is licensed under the repository license

### Notes for reviewers

Only the two executable references required by the runtime contract are changed: `references/visual-changenet.md` and `references/prepare-for-inference.md`. No extra note or planning Markdown files are added. The existing C-RADIO profile remains the default. For DINOv3, the baseline spec is the source of truth and `freeze_backbone: true` is enforced before staging or training.

### CODE REVIEW

File: skills/applications/tao-run-deft-aoi/scripts/stage_backbone.py [hunk 1]
```diff
@@ -70,17 +123,122 @@ def parse_args() -> argparse.Namespace:
...
+    backbone = _baseline_backbone(args)
+    baseline_type = str(backbone.get("type", "")).strip()
+    backbone_type = (
+        getattr(args, "backbone_type", None)
+        or baseline_type
+        or DEFAULT_BACKBONE_TYPE
+    )
+
+    if backbone_type not in BACKBONE_PROFILES:
+        sys.exit(f"stage_backbone: unsupported backbone type: {backbone_type}")
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/scripts/stage_backbone.py:153]: Question: `main()` now always infers the profile from `baseline_spec.yaml`, so a workspace using a backbone outside these 7 profiles hard-stops here — even when the caller passes the documented `--dest`/`--repo-id`/`--filename` escape hatches. Intended? If so, maybe let explicit `--repo-id`/`--dest` bypass profile inference so the custom-repo recipe in the docstring keeps working.

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py [hunk 2]
```diff
@@ -139,6 +139,86 @@ def _build_inference_spec(
...
+    train_spec: dict[str, Any],
+    state: dict[str, Any],
+) -> tuple[str, str, str, bool]:
+    """Resolve the host backbone while preserving the legacy AOI mount."""
+    backbone_config = train_spec.get("model", {}).get("backbone", {})
+    container_path = backbone_config.get("pretrained_backbone_path")
+    backbone_type = str(backbone_config.get("type", ""))
+    backbone_frozen = bool(backbone_config.get("freeze_backbone", False))
+    if not isinstance(container_path, str) or not container_path.strip():
+        if "dinov3" in backbone_type and backbone_frozen:
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: When is this branch reachable? Pre-Flight both stages the file and rewrites `pretrained_backbone_path` in the specs, so "path empty but HF file staged" implies training ran without these weights — yet the handoff points the inference spec at them. Safer to hard-error on a missing path unconditionally, or at least log loudly that a fallback is being substituted?

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py [hunk 3]
```diff
@@ -139,6 +139,86 @@ def _build_inference_spec(
...
+            aliases = [
+                path
+                for path in candidates
+                if "".join(
+                    char for char in path.stem.casefold() if char.isalnum()
+                ) == configured_stem
+            ]
+            if len(aliases) == 1:
+                host_path = aliases[0]
+            elif len(candidates) == 1:
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: This fallback can silently hand off a mismatched weight: if the expected C-RADIO file is missing but exactly one other weight sits in `backbone/` (e.g. a leftover `vit_large_dinov3.safetensors`), it gets selected and recorded at the C-RADIO container path. Should it require an alias match and otherwise fall through to the `FileNotFoundError` below?

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py [hunk 4]
```diff
@@ -139,6 +139,86 @@ def _build_inference_spec(
...
+) -> tuple[str, str, str, bool]:
+    """Resolve the host backbone while preserving the legacy AOI mount."""
+    backbone_config = train_spec.get("model", {}).get("backbone", {})
+    container_path = backbone_config.get("pretrained_backbone_path")
+    backbone_type = str(backbone_config.get("type", ""))
+    backbone_frozen = bool(backbone_config.get("freeze_backbone", False))
+    if not isinstance(container_path, str) or not container_path.strip():
+        if "dinov3" in backbone_type and backbone_frozen:
+            # The DEFT pre-flight stages the matching HF file before TAO runs.
+            stage_name = f"{backbone_type}.safetensors"
```
Comment (vpraveen-nv) [inline skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Nit: this re-derives the `<type>.safetensors` convention that `stage_backbone.BACKBONE_PROFILES` owns — a lookup there would keep one source of truth if a profile's stage name ever diverges.

**Review by vpraveen-nv — APPROVED** [review]
LGTM — precedence and hard-stop semantics match the description and are well tested (including the decoy-file case), schemas align with tao-core#31, SegFormer is untouched, and C-RADIO stays the default. A few non-blocking questions inline.

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py (hunk 3 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Addressed in 320c787. The unrelated sole-file fallback is removed. Legacy C-RADIO renames now require one normalized alias match; otherwise handoff raises the existing exact-backbone FileNotFoundError. Added a mismatched-sole-file regression test.

File: skills/applications/tao-run-deft-aoi/scripts/stage_backbone.py (hunk 1 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/scripts/stage_backbone.py:153]: Addressed in 320c787. Explicit dest, repo-id, or filename overrides now bypass workspace profile inference when no backbone type is explicitly selected, preserving the documented custom-repo escape hatch. Added a workspace regression test with an unsupported customer backbone profile.

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py (hunk 4 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Addressed in 320c787 by removing the empty-path fallback entirely. prepare_inference_spec no longer re-derives a type-based stage filename, so stage_backbone remains the sole owner of profile stage names.

File: skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py (hunk 2 above)
Comment (nnagrajrao) [reply skills/applications/tao-run-deft-aoi/scripts/prepare_inference_spec.py]: Addressed in 320c787. Missing or empty pretrained_backbone_path now hard-errors unconditionally during handoff; inference no longer substitutes staged weights that training did not explicitly reference. Added a regression test.

**Comment by christinayyw** [comment]
## TAO Skill Execution Eval Summary

**Latest run:** [tao-sb-OSS-develop #140](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/140/) — `320c787a` @ 2026-08-11 14:13 UTC

| Skill | Eval Case | Backend | Verdict | Report |
|---|---|---|---|---|
| `skills/applications/tao-run-deft-aoi` | — | `claude` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/140/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #139 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/139/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html)<br>[Build #138 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/138/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-claude/skill-eval-report.html)</details> |
| `skills/applications/tao-run-deft-aoi` | — | `codex` | ✅ Pass (⚠️ w/ self-correction) | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/140/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #139 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/139/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html)<br>[Build #138 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/138/artifact/jenkins-eval-reports/skills-applications-tao-run-deft-aoi-codex/skill-eval-report.html)</details> |
| `skills/models/tao-train-visual-changenet` | — | `claude` | ✅ Pass | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/140/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-claude/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #139 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/139/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-claude/skill-eval-report.html)<br>[Build #138 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/138/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-claude/skill-eval-report.html)</details> |
| `skills/models/tao-train-visual-changenet` | — | `codex` | ✅ Pass | [Latest Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/140/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-codex/skill-eval-report.html)<br><details><summary>Previous Reports</summary>[Build #139 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/139/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-codex/skill-eval-report.html)<br>[Build #138 Eval Report](https://prod.blsm.nvidia.com/metropolis-iva-tlt-jenkins/job/tao-sb-OSS-develop/138/artifact/jenkins-eval-reports/skills-models-tao-train-visual-changenet-codex/skill-eval-report.html)</details> |


<sub>✅ Pass = expectation met without self-correction &nbsp;·&nbsp; ✅ Pass (⚠️ w/ self-correction) = expectation met but the agent rewrote / retried &nbsp;·&nbsp; ❌ Fail = expectation not met &nbsp;·&nbsp; ❓ Error = shard didn't produce an eval report</sub>
